tests/virt/upgrade/conftest.py

```diff
@@ -156,24 +151,26 @@ def upgrade_namespaces(upgrade_namespace_scope_session, kmp_enabled_namespace):
 
 
 @pytest.fixture(scope="session")
-def migratable_vms(admin_client, hco_namespace, upgrade_namespaces):
-    migratable_vms = get_all_migratable_vms(admin_client=admin_client, namespaces=upgrade_namespaces)
+def migratable_vms(admin_client, upgrade_namespaces):
+    all_vms_list = [
+        vm for ns in upgrade_namespaces for vm in list(VirtualMachine.get(client=admin_client, namespace=ns.name))
+    ]
+    migratable_vms = [
+        vm
+        for vm in all_vms_list
+        if any(
```

Comment:
how about combining it? also I think it may win in readability, somethig like:

```
    migratable_vms = []
    for ns in upgrade_namespaces:
        for vm in list(VirtualMachine.get(client=admin_client, namespace=ns.name):
            if any(condition.type ...)
                migratable_vms.append(vm)
```